packages/lab/stories/date-picker/date-picker.stories.tsx

```diff
@@ -1,209 +1,1099 @@
+  const helperText = "Date format DD MMM YYYY (e.g. 09 Jun 2024)";
+  const minDate = today(getLocalTimeZone());
+  return (
+    <FormField>
+      <FormLabel>Select a date range</FormLabel>
+      <DatePicker
+        {...args}
+        selectionVariant="range"
+        minDate={minDate}
+        maxDate={minDate.add({ years: 50 })}
+        onSelectedDateChange={(newSelectedDate) => {
+          console.log(
+            `Selected date range: ${formatDateRange(newSelectedDate)}`,
+          );
+          args?.onSelectedDateChange?.(newSelectedDate);
+        }}
+      >
+        <DatePickerRangeInput />
+        <DatePickerOverlay>
+          <FlexLayout gap={0} direction="column">
+            <FlexItem>
+              <DatePickerRangePanel helperText={helperText} />
+              <Divider variant="tertiary" />
+            </FlexItem>
+            <FlexItem>
+              <DatePickerActions selectionVariant="range" />
+            </FlexItem>
+          </FlexLayout>
+        </DatePickerOverlay>
+      </DatePicker>
+      <FormHelperText>{helperText}</FormHelperText>
+    </FormField>
+  );
+};
+
+export const SingleWithCustomParser: StoryFn<DatePickerSingleProps> = (
+  args,
+) => {
+  const helperText =
+    "Date format DD MMM YYYY (e.g. 09 Jun 2024) or +/-D (e.g. +7)";
+  const [validationStatus, setValidationStatus] = useState<"error" | undefined>(
+    undefined,
+  );
+  const [selectedDate, setSelectedDate] = useState<SingleDateSelection | null>(
+    null,
+  );
+
+  return (
+    <FormField validationStatus={validationStatus}>
+      <FormLabel>Select a date</FormLabel>
+      <DatePicker
+        {...args}
+        selectionVariant="single"
+        selectedDate={selectedDate}
+        onSelectedDateChange={(newSelectedDate: SingleDateSelection | null) => {
+          console.log(`Selected date: ${formatDate(newSelectedDate)}`);
+          setSelectedDate(newSelectedDate);
+          setValidationStatus(undefined);
+          args?.onSelectedDateChange?.(newSelectedDate);
+        }}
+      >
+        <DatePickerSingleInput
+          parse={(inputDate: string | undefined): DateValue | undefined => {
+            const parsedDate = inputDate;
+            const offsetMatch = parsedDate?.match(/^([+-]?\d+)$/);
+            if (offsetMatch) {
+              const offsetDays = Number.parseInt(offsetMatch[1], 10);
+              let offsetDate = selectedDate
+                ? selectedDate
+                : today(getLocalTimeZone());
+              offsetDate = offsetDate.add({ days: offsetDays });
+              return new CalendarDate(
+                offsetDate.year,
+                offsetDate.month,
+                offsetDate.day,
+              );
+            }
+            return parseCalendarDate(parsedDate);
+          }}
+          onDateValueChange={(newDateValue: string) => {
+            const validationStatus =
+              isValidShortDate(newDateValue) ||
```

Comment:
`validateShortDate` is making it error again after "+7" correctly parsed

Reply:
fixed now, was a short-coming in the example code